src/routes/safe/components/DropdownCurrency/index.jsx

```diff
@@ -0,0 +1,239 @@
+    padding: '0',
+    textAlign: 'left',
+    '&:active': {
+      opacity: '0.8',
+    },
+  },
+  buttonInner: {
+    boxSizing: 'border-box',
+    display: 'block',
+    height: '100%',
+    lineHeight: '24px',
+    padding: '0 22px 0 8px',
+    position: 'relative',
+    width: '100%',
+    '&::after': {
+      borderLeft: '5px solid transparent',
+      borderRight: '5px solid transparent',
+      borderTop: '5px solid #5d6d74',
+      content: '""',
+      height: '0',
+      position: 'absolute',
+      right: '8px',
+      top: '9px',
+      width: '0',
+    },
+  },
+  openMenuButton: {
+    '&::after': {
+      borderBottom: '5px solid #5d6d74',
+      borderLeft: '5px solid transparent',
+      borderRight: '5px solid transparent',
+      borderTop: 'none',
+    },
+  },
+  dropdownItemsScrollWrapper: {
+    maxHeight: '280px',
+    overflow: 'auto',
+  },
+  search: {
+    position: 'relative',
+    borderRadius: '0',
+    backgroundColor: '#fff',
+    '&:hover': {
+      backgroundColor: '#fff',
+    },
+    marginRight: 0,
+    width: '100%',
+  },
+  searchIcon: {
+    alignItems: 'center',
+    display: 'flex',
+    height: '100%',
+    justifyContent: 'center',
+    left: '12px',
+    margin: '0',
+    pointerEvents: 'none',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translateY(-50%)',
+    width: '18px',
+    '& path': {
+      fill: '#b2b5b2',
+    },
+  },
+  inputRoot: {
+    color: '#5d6d74',
+    fontSize: '14px',
+    fontWeight: 'normal',
+    lineHeight: '1.43',
+    width: '100%',
+  },
+  inputInput: {
+    boxSizing: 'border-box',
+    height: '44px',
+    padding: '12px 12px 12px 40px',
+    width: '100%',
+  },
+})
+
+const DropdownCurrency = () => {
+  const currenciesList = Object.values(AVAILABLE_CURRENCIES)
+  const dispatch = useDispatch()
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+  const currencyValueSelected = useSelector(currentCurrencySelector)
+
+  const [searchParams, setSearchParams] = useState('')
+  const classes = useStyles()
+  const currenciesListFiltered = currenciesList.filter((currency) => currency.toLowerCase().includes(searchParams.toLowerCase()))
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
+  const onCurrentCurrencyChangedHandler = (newCurrencySelectedName: AVAILABLE_CURRENCIES) => {
+    dispatch(fetchCurrencySelectedValue(newCurrencySelectedName))
+    dispatch(setCurrencySelected(newCurrencySelectedName))
+    handleClose()
+  }
+
+
+  return (
+    !currencyValueSelected ? null
+      : (
+        <MuiThemeProvider theme={DropdownListTheme}>
+          <>
+            <button
+              className={classes.button}
+              onClick={handleClick}
+              type="button"
+            >
+              <span className={`${classes.buttonInner} ${anchorEl ? classes.openMenuButton : ''}`}>
```

Comment:
you can use `classnames` package from our dependency list

Reply:
What is the advantage of using that package?, I think it won't make it simpler

Reply:
@Agupane Imo `cn(classes.buttonInnter, anchorEl && classes.openMenuButton)` is easier to read. Also we use it across the project

Reply:
done